### PR TITLE
Add descriptions and joint parties

### PR DESCRIPTION
**Pull request by mhl**
This pull request:

* Incorporates party descriptions into PopIt and the JSON dump
  of party information
* Adds pseudo parties for join parties

### CODE REVIEW

File: candidates/management/commands/candidates_create_joint_pseudo_parties.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+def create_or_update_party(api, joint_party_name, sub_parties):
+    party_ids = sorted(
+        [p['id'] for p in sub_parties],
+        key=lambda party_id: extract_number_from_id(party_id)
+    )
+    registers = set(p['register'] for p in sub_parties)
+    if len(registers) > 1:
+        raise Exception("Multiple registers found " + repr(registers))
+    joint_party_id = '-'.join(party_ids)
```
Comment (tfgg) [inline candidates/management/commands/candidates_create_joint_pseudo_parties.py]: Small thought: Should this be fixed with "party:" or "joint-party:" or similar to be consistent with the other party ids?

File: candidates/management/commands/candidates_create_joint_pseudo_parties.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        else:
+            raise
+
+joint_description_re = re.compile(
+    r'^(?P<joint_name>.*?) \([jJ]oint [dD]escription with (?P<sub_party>.*)\)'
+)
+
+
+class Command(BaseCommand):
+    help = "Update parties from a CSV of party data"
```
Comment (tfgg) [inline candidates/management/commands/candidates_create_joint_pseudo_parties.py]: Help text could explain briefly what's going on here?

File: candidates/management/commands/candidates_create_joint_pseudo_parties.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        api.organizations.post(party_data)
+    except HttpServerError as e:
+        if 'E11000' in e.content:
+            # Duplicate Party Found
+            api.organizations(joint_party_id).put(party_data)
+        else:
+            raise
+
+joint_description_re = re.compile(
+    r'^(?P<joint_name>.*?) \([jJ]oint [dD]escription with (?P<sub_party>.*)\)'
```
Comment (tfgg) [inline candidates/management/commands/candidates_create_joint_pseudo_parties.py]: the "Left Unity - Trade Unionists and Socialists" is being missed on the "Left Unity" side due to an extra s:

Left Unity - Trade Unionists and Socialists (Joint Descriptions with Trade Unionists and Socialists Coalition)

**Comment by mhl** [comment]
@tfgg Thanks for the comments, I think I've fixed all those things now

**Comment by tfgg** [comment]
:+1:

**Pull request merged**